pkg/network/controllers/vm_test.go

```diff
@@ -509,6 +509,70 @@ var _ = Describe("VM Network Controller", func() {
 		Expect(iface).NotTo(BeNil())
 		Expect(iface.State).NotTo(Equal(v1.InterfaceStateAbsent))
 	})
+
+	It("Sync successfully performs hotplug and hotunplug at the same time", func() {
+		const (
+			netToAttachName = "attach-me"
+
+			netToDetachName    = "detach-me"
+			netToDetachNADName = "detach-me-nad"
+		)
+
+		clientset := fake.NewSimpleClientset()
+		c := controllers.NewVMController(clientset)
+
+		multusAndDomainInfoSource := vmispec.NewInfoSource(vmispec.InfoSourceMultusStatus, vmispec.InfoSourceDomain)
+
+		ifaceToDetach := libvmi.InterfaceDeviceWithBridgeBinding(netToDetachName)
+		netToDetach := libvmi.MultusNetwork(netToDetachName, netToDetachNADName)
+
+		vmi := libvmi.New(
+			libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+			libvmi.WithInterface(ifaceToDetach),
+			libvmi.WithNetwork(v1.DefaultPodNetwork()),
+			libvmi.WithNetwork(netToDetach),
+			libvmistatus.WithStatus(
```

Comment:
this was not in the original test, can you explain why it is needed?

Reply:
The original test only tested a small fraction of the logic.
The proposed test, tests the whole VM controller's logic in the scenario when one interface is hotplugged and another is hotunplugged.
In the test, we hotunplug a device that is currently plugged (has an interface status with certain info sources),